module/VuFind/tests/integration-tests/src/VuFindTest/Mink/EmailAuthenticationTest.php

```diff
@@ -0,0 +1,130 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+
+namespace VuFindTest\Mink;
+
+/**
+ * Email authentication test class.
+ *
+ * Class must be final due to use of "new static()" by LiveDatabaseTrait.
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Page
+ */
+final class EmailAuthenticationTest extends \VuFindTest\Integration\MinkTestCase
+{
+    use \VuFindTest\Feature\EmailTrait;
+    use \VuFindTest\Feature\LiveDatabaseTrait;
+
+    /**
+     * Standard setup method.
+     *
+     * @return void
+     */
+    public static function setUpBeforeClass(): void
+    {
+        static::failIfDataExists();
+    }
+
+    /**
+     * Test the email authentication process.
+     *
+     * @return void
+     */
+    public function testEmailAuthentication(): void
+    {
+        // Set up configs, session and message logging:
+        $this->changeConfigs(
+            [
+                'config' => [
+                    'Authentication' => [
+                        'method' => 'ILS',
+                    ],
+                    'Catalog' => [
+                        'driver' => 'Demo',
+                    ],
+                    'Mail' => [
+                        'testOnly' => true,
+                        'message_log' => $this->getEmailLogPath(),
+                        'default_from' => 'noreply@vufind.org',
+                    ],
+                ],
+                'Demo' => [
+                    'Catalog' => [
+                        'loginMethod' => 'email',
+                    ],
+                ],
+            ]
+        );
+
+        $this->resetEmailLog();
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl());
+        $page = $session->getPage();
+
+        // Request login:
+        $this->clickCss($page, '#loginOptions a');
+        $this->findCssAndSetValue($page, '.modal-body [name="username"]', 'catuser@vufind.org');
+        $this->clickCss($page, '.modal-body .btn.btn-primary');
+        $this->assertEquals(
+            'We have sent a login link to your email address. It may take a few moments for the link to arrive.'
+            . " If you don't receive the link shortly, please check also your spam filter.",
+            $this->findCssAndGetText($page, '.alert-success')
+        );
+
+        // Extract the link from the provided message:
+        $email = file_get_contents($this->getEmailLogPath());
```

Comment:
I assume you'll want to update this to match the improved behavior in #3975. Would it make sense to pull this test out to a separate PR so we don't lose track, since nothing else here really depends on it? Or should we just add a TODO checkbox somewhere?

Reply:
...and now that #3975 is merged, you can just address it here if you wish. :-)

Reply:
Done!